ts/backbone/Conversation.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @prettier
+ */
+import is from '@sindresorhus/is';
+
+import { Collection as BackboneCollection } from '../types/backbone/Collection';
+import { deferredToPromise } from '../../js/modules/deferred_to_promise';
+import { Message } from '../types/Message';
+
+export const fetchVisualMediaAttachments = async ({
+  conversationId,
+  WhisperMessageCollection,
+}: {
+  conversationId: string;
+  WhisperMessageCollection: BackboneCollection<Message>;
+}): Promise<Array<Message>> => {
+  if (!is.string(conversationId)) {
+    throw new TypeError("'conversationId' is required");
+  }
+
+  if (!is.object(WhisperMessageCollection)) {
+    throw new TypeError("'WhisperMessageCollection' is required");
+  }
+
+  const collection = new WhisperMessageCollection();
+  const lowerReceivedAt = 0;
+  const upperReceivedAt = Number.MAX_VALUE;
+  const hasVisualMediaAttachments = 1;
+  await deferredToPromise(
+    collection.fetch({
+      index: {
+        name: 'hasVisualMediaAttachments',
+        lower: [conversationId, hasVisualMediaAttachments, lowerReceivedAt],
+        upper: [conversationId, hasVisualMediaAttachments, upperReceivedAt],
```

Comment:
What made you decide to have the boolean element be second in the key array, vs. last? Also, do we expect to have messages without visual media attachments in this index? (essentially: it it `1` and nothing, or `1` and `0`?)